fastplotlib/layouts.py

```diff
@@ -7,6 +7,12 @@
 from wgpu.gui.auto import WgpuCanvas
 
 
+class GridPlotIndexer():
+    def __init__(self):
+        self.name: str = None
+        self.index: tuple = None
+
```

Comment:
I think `self.subplot` should be an attribute too, and this would contain the actual subplot.

We should then think about whether to keep the current API of:

```python
gp.subplots[i, j]  # returns subplot at (i, j)

# since this would basically be a duplicate of

gp[i, j].subplot  # returns subplot at (i, j)

but this GridPlotIndexer() interface provides more features like
gp[i, j].name  # returns a user-set name etc.
```

I think we could just make `gp.subplots` a private attribute that stores the subplots as a numpy array, and `gp[i, j].subplot` just calls the appropriate one from private `gp._subplots` which is a numpy array of subplot objects.

Reply:
@clewis7 you had a question about this today, take a look at it again and if you have questions I will clarify Friday.

Reply:
We could also deprecate `GridPlot.subplots` and just make the subplots an attribute of `GridPlotIndexer` instead.